app/models/container.rb

```diff
@@ -27,16 +27,17 @@ class Container < ApplicationRecord
 
   acts_as_miq_taggable
 
-  def event_where_clause(assoc = :ems_events)
-    case assoc.to_sym
-    when :ems_events, :event_streams
-      # TODO: improve relationship using the id
-      ["container_namespace = ? AND #{events_table_name(assoc)}.ems_id = ? AND container_name = ?",
-       container_project.try(:name), ext_management_system.try(:id), name]
-    when :policy_events
-      # TODO: implement policy events and its relationship
-      ["#{events_table_name(assoc)}.ems_id = ?", ext_management_system.try(:id)]
-    end
+  def event_where_clause_ems_events
+    # TODO: improve relationship using the id
+    EmsEvent
+      .where(:container_name => name)
+      .where(:container_namespace => container_project.try(:name))
+      .where(:ems_id => ext_management_system.try(:id))
+  end
+
+  def event_where_clause_miq_events
+    # TODO: implement policy events and its relationship
+    MiqEvent.where(:ems_id => ext_management_system.try(:id))
```

Comment:
is this a `has_many :miq_events` with a foreign key of `ems_id`?
not sure if the caching of events would help or harm us though

Reply:
Container doesn't even have a relationship to it.  It's in the list of weird behaving classes in the description.

[![image](https://user-images.githubusercontent.com/19339/146990478-7746645a-069b-4590-9f71-5fb04555a9dc.png)
](https://github.com/ManageIQ/manageiq/pull/21584#issue-1069869129)